get_openstack_info_open_port/get_openstack_info_open_port/get_openstack_info_open_port.py

```diff
@@ -0,0 +1,262 @@
+        self.config['log_file'] = log_file
+        self.ip = ip
+        self.port = port
+        self.rule_found = False
+        self._init_log()
+        self._os_auth()
+        self.all_floating_ips = self.neutron.list_floatingips(floating_ip_address=self.ip)['floatingips']
+        self._log.debug(f"Found {len(self.all_floating_ips)} floating IPs.")
+        if self._check_known_ip():
+            self._log.debug('IP found in your environment.')
+            self._test_open_port()
+            self._check_openstack()
+            if method == 'IP':
+                self.floatingip = self._get_openstack_floating_ip()
+                port = self.neutron.show_port(self.floatingip['port_id'])['port']
+                for security_group in port['security_groups']:
+                    self._check_security_group(security_group)
+            else:
+                self._log.debug(f"Looking for instance with IP '{ip}' in OpenStack's region '{os.environ['OS_REGION_NAME']}' as user '{os.environ['OS_USERNAME']}' with port '{port}' open to incoming connections...")
+                self._get_openstack_instances()
+                self._locate_affected_instance()
+                self._check_affected_security_groups()
+            if not self.rule_found:
+                self._log.info(f"No rule found that allow all incoming traffic to IP ̣{self.ip} and port {self.port}")
+                sys.exit(5)
+        else:
+            self._log.info(f"The IP '{self.ip}' couldn't be found in your OpenStack environment.")
+            sys.exit(6)
+            
+    def _check_known_ip(self):
+        for ip in self.all_floating_ips:
+            if self.ip == ip['floating_ip_address']:
+                return True
+        return False
+        
+    def _check_security_group(self, security_group_id):
+        self._check_security_group_rules(self.neutron.show_security_group(security_group_id)['security_group']['security_group_rules'])
+
+    def _check_security_group_rules(self, security_group_rules):
+        for rule in security_group_rules:
+            if rule.get('direction', '') == 'ingress':
+                remote_ip_prefix = rule.get('remote_ip_prefix', '')
+                if remote_ip_prefix:
+                    netmask = remote_ip_prefix.split("/")[1]
+                    if netmask == '0':
+                        port_range_max = rule.get('port_range_max', 0)
+                        if port_range_max is not None and port_range_max <= self.port:
+                            port_range_min = rule.get('port_range_min', 0)
+                            if port_range_min is not None and port_range_min >= self.port:
+                                self._log.warn(f"Attention! Rule '{rule['id']}' in security group '{rule['security_group_id']}' allows incoming traffic from everywhere.")
+                                self.rule_found = True
+                        else:
+                            print('no match')
```

Comment:
I have a form observation regarding this potion of the code. I do not like to make if inside if. As solution, I can suggest the following replace the above code with the above code:
    def _check_security_group_rules(self, security_group_rules):
        for rule in security_group_rules:
            remote_ip_prefix = rule.get('remote_ip_prefix', '')
            if rule.get('direction', '') != 'ingress' or not remote_ip_prefix:
               continue

            netmask = remote_ip_prefix.split("/")[1]
            if netmask != '0':
                continue
            port_range_max = rule.get('port_range_max', 0) 
            port_range_min = rule.get('port_range_min', 0)  
            if port_range_max is not None and and port_range_min is not None and port_range_max <= self.port  and port_range_min >= self.port:
               self._log.warn(f"Attention! Rule '{rule['id']}' in security group '{rule['security_group_id']}' allows incoming traffic from everywhere.")
               self.rule_found = True
           else:
                 print('no match')

Reply:
Done